contracts/YieldCrabLever.sol

```diff
@@ -0,0 +1,451 @@
+            msg.sender,
+            balances.ink,
+            balances.art
+        );
+    }
+
+    /// @notice Divest, either before or after maturity.
+    /// @param vaultId The vault to divest from.
+    /// @param seriesId The series
+    /// @param ilkId The ilkId
+    /// @param ink The amount of collateral to recover.
+    /// @param art The amount of debt to repay.
+    /// @param minBaseOut Used to minimize slippage. The transaction will revert
+    ///     if we don't obtain at least this much of the base asset.
+    function divest(
+        bytes12 vaultId,
+        bytes6 seriesId,
+        bytes6 ilkId,
+        uint256 ink,
+        uint256 art,
+        uint256 minBaseOut
+    ) external {
+        // Test that the caller is the owner of the vault.
+        // This is important as we will take the vault from the user.
+        require(cauldron.vaults(vaultId).owner == msg.sender);
+
+        // Give the vault to the contract
+        giver.seize(vaultId, address(this));
+
+        // Check if we're pre or post maturity.
+        bool success;
+        if (uint32(block.timestamp) > cauldron.series(seriesId).maturity) {
+            bytes memory data = bytes.concat(
+                bytes1(bytes1(uint8(uint256(Operation.CLOSE)))), // [0:1]
+                seriesId, // [1:7]
+                vaultId, // [7:19]
+                ilkId, // [19:25]
+                bytes32(ink), // [25:57]
+                bytes32(art)
+            );
+
+            address join = address(ladle.joins(seriesId & ASSET_ID_MASK));
+
+            // Close:
+            // Series is not past maturity, borrow and move directly to collateral pool.
+            // We have a debt in terms of fyToken, but should pay back in base.
+            uint128 base = cauldron.debtToBase(seriesId, art.u128());
+            success = IERC3156FlashLender(join).flashLoan(
+                this, // Loan Receiver
+                address(IJoin(join).asset()), // Loan Token
+                base, // Loan Amount
+                data
+            );
+        } else {
+            IPool pool = IPool(ladle.pools(seriesId));
+            IMaturingToken fyToken = pool.fyToken();
+            // Repay:
+            // Series is not past maturity.
+            // Borrow to repay debt, move directly to the pool.
+            bytes memory data = bytes.concat(
+                bytes1(bytes1(uint8(uint256(Operation.REPAY)))), // [0:1]
+                seriesId, // [1:7]
+                vaultId, // [7:19]
+                ilkId, // [19:25]
+                bytes32(ink), // [25:57]
+                bytes32(art) // [57:89]
+            );
+            success = IERC3156FlashLender(address(fyToken)).flashLoan(
+                this, // Loan Receiver
+                address(fyToken), // Loan Token
+                art, // Loan Amount: borrow exactly the debt to repay.
+                data
+            );
+            require(IERC20(address(fyToken)).balanceOf(address(this)) == 0);
```

Comment:
Why this require? if there is a rounding error or something we could brick the contract. What are we trying to avoid here?